netket/machine/jax.py

```diff
@@ -149,16 +148,20 @@ def vector_jacobian_prod(
                 out = tree_map(jax.numpy.conjugate, pout[0])
 
             return out
+
         else:
+
             if conjugate and self._dtype is complex:
-                prodj = lambda j: jax.np.tensordot(vec, j.conjugate(), axes=vec.ndim)
+                prodj = lambda j: jax.np.tensordot(vec.transpose(), j.conjugate(), axes=1)
             else:
-                prodj = lambda j: jax.np.tensordot(vec.conjugate(), j, axes=vec.ndim)
+                prodj = lambda j: jax.np.tensordot(vec.transpose().conjugate(), j, axes=1)
```

Comment:
was the old version with axes=vec.ndim wrong?

Reply:
Yes, it didn't work. Axes are the dimensions to contract but "1" contracts the last index of the first arg with the first index of the second arg. Since j takes on different shapes, this seemed like the most straightforward solution